docs/system-admin-guide/integrations/nextcloud/README.md

```diff
@@ -1,100 +1,183 @@
+
+#### 3. Enter OpenProject OAuth values in Nextcloud settings
+
+At this point, you will see a page titled **OpenProject OAuth application details**.
+
+Note that OpenProject has automatically generated an OAuth **client ID** and a **client secret**. These values are needed to permit Nextcloud to connect to OpenProject.
+
+> **Important**: These generated values are not accessible again after you close the window. Please do not navigate away from this page before copying them over to Nextcloud, as instructed below. Treat these values with care, as you would an important password. Please do not reveal them to anyone else.
+
+
+
+![OpenProject generates OAuth values to copy over to Nextcloud](OP-OAuth-values.png)
+
+Go back to the browser tab where you were configuring the **OpenProject integration** app. (We recommend you have two browser tabs open: the current one with OpenProject and the former one with Nextcloud).
+
+Copy the two generated values (client ID and secret) from the OpenProject tab to the respective fields in Nextcloud, namely **OpenProject OAuth client ID** and **OpenProject OAuth client secret**. 
+
+![OAuth values generated by OpenProject are entered into Nextcloud app configuration](3_2_03-NC_Step_2.png)
+
+Once you have copied the values, click on **Save** to proceed to the next step.
+
+#### 4. Enter Nextcloud OAuth values in OpenProject
+
+In the page that appears, you will see new OAuth values that are once again generated automatically, but this time by Nextcloud.
+
+OpenProject will need these values to be able to connect to your Nextcloud instance.
+
+Much like in the previous step, you will need to copy these two generated values (**Nextcloud OAuth client ID** and **Nextcloud OAtuh client secret**) and paste them into OpenProject.
+
+![Nextcloud also generates OAuth values that need to be copied to OpenProject](3_2_04-NC_Step_3.png)
+
+Navigate back to your OpenProject tab and click on the **Done. Continue setup** button on the screen you previously left it at. You will now see a screen where you will be able to enter the Nextcloud values. 
 
 ![OpenProject_NC_OAuth_values](3_3_01-OP_OAuth_application_details.png)
 
-Once these values are entered, you can click the button **Save and complete setup** in your OpenProject tab and the **Yes, I have copied these values** in Nextcloud. Once this is done, your instance configuration is complete.
+Once you hava entered the client ID and client secrets on this page, click on **Save and complete setup**. In the next screen, click on **Yes, I have copied these values**. At this point, your instance configuration is complete and you should see a green banner confirming this.
+
+![Integration successfully completed on the OpenProject end](/Users/parimal/Documents/GitHub/openproject/docs/system-admin-guide/integrations/nextcloud/Nextcloud-set-up-in-OP.png)
```

Comment:
Please remove local user path
Change it to: (Nextcloud-set-up-in-OP.png)

Reply:
Done.